src/movies.js

```diff
@@ -1,16 +1,112 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+let getAllDirectors = movies.map(function(movie){
+  return movie.director;
+});
+
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+
+  let howManyMovies= movies.filter(function(movie){
+    if (movie["genre"].includes("Drama") && movie["director"] === "Gary Trousdale")
+       return movie;
+   });
+
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+
+let ratesTotal = movies.reduce(function(acc, movie){
+  return acc + movie.rate
+},0);
+
+console.log((ratesTotal / movies.length).toFixed(2));
```

Comment:
toFixed returns a string. You would need to do something else to return a number instead ;)